Source/Plugins/Core/com.equella.core/js/tsrc/mainui/index.tsx

```diff
@@ -58,216 +27,15 @@ declare const renderData:
-            <LegacyPage
-              {...mkRouteProps(p)}
-              errorCallback={errorCallback}
-              legacyContent={{ content, setLegacyContentProps }}
-            />
-          )}
-        />
-      </Switch>
-    );
-  }
-
-  return (
-    <BrowserRouter
-      basename={basePath}
-      getUserConfirmation={(message, cb) => {
-        if (errorShowing.current) {
-          errorShowing.current = false;
-          setFullPageError(undefined);
-          cb(true);
-        } else {
-          setNavAwayCallback({ message, cb });
-        }
-      }}
-    >
-      <Prompt
-        when={Boolean(preventNavMessage) || errorShowing.current}
-        message={nonBlankNavMsg}
-      />
-      <NavAwayDialog
-        open={Boolean(navAwayCallback)}
-        message={nonBlankNavMsg}
-        navigateConfirm={(confirm) => {
-          if (navAwayCallback) navAwayCallback.cb(confirm);
-          if (confirm) setPreventNavMessage(undefined);
-          setNavAwayCallback(undefined);
-        }}
-      />
-      <LegacyContent
-        {...legacyContentProps}
-        render={(content) => {
-          const tp = content
-            ? templatePropsForLegacy(content)
-            : {
-                ...templateProps,
-                fullscreenMode: legacyContentProps.enabled
-                  ? templateProps.fullscreenMode
-                  : undefined,
-              };
-          const withErr = fullPageError
-            ? { ...tp, title: fullPageError.error, fullscreenMode: undefined }
-            : tp;
-          const template = (
-            <Template {...withErr} currentUser={currentUser}>
-              {routeSwitch(content)}
-            </Template>
-          );
-          const render = () => {
-            if (!content || content.noForm) {
-              return template;
-            } else {
-              const { form } = content.html;
-              return (
-                <>
-                  <LegacyForm state={content.state}>{template}</LegacyForm>
-                  {form && HtmlParser(form)}
-                </>
-              );
-            }
-          };
-          return render();
-        }}
-      />
-    </BrowserRouter>
-  );
-}
+const App = React.lazy(() => import("./App"));
 
 export default function () {
   initStrings();
-  if (typeof renderData !== "undefined") {
-    startHeartbeat();
-    ReactDOM.render(
-      <ThemeProvider theme={oeqTheme}>
-        <IndexPage />
-      </ThemeProvider>,
-      document.getElementById("mainDiv")
-    );
-  } else {
-    ReactDOM.render(
-      <BrowserRouter basename={basePath} forceRefresh>
-        <ThemeProvider theme={oeqTheme}>
-          <SettingsPage refreshUser={() => {}} updateTemplate={() => {}} />
-        </ThemeProvider>
-      </BrowserRouter>,
-      document.getElementById("settingsPage")
-    );
-  }
+  const legacySettingsMode = typeof renderData === "undefined";
+  ReactDOM.render(
+    <React.Suspense fallback={null}>
```

Comment:
No fallback content?
Would it have value to show something, even the text loading?

```suggestion
    <React.Suspense fallback={<>loading</>}>
```